mobile_wallet/src/lib.rs

```diff
@@ -766,6 +768,73 @@ fn decrypt_encrypted_amount_aux(input: &str) -> anyhow::Result<Amount> {
     )
 }
 
+fn parse_wallet_input(v: &Value) -> anyhow::Result<ConcordiumHdWallet> {
+    let seed_hex: String = try_get(v, "seed")?;
+    let net: Net = try_get(v, "net")?;
+    let wallet = ConcordiumHdWallet {
+        seed: hex::decode(&seed_hex)?,
+        net,
+    };
+    Ok(wallet)
+}
+
+fn get_identity_keys_and_randomness_aux(input: &str) -> anyhow::Result<String> {
+    let v: Value = from_str(&input)?;
+    let wallet = parse_wallet_input(&v)?;
+    let identity_index = try_get(&v, "identityIndex")?;
+
+    let id_cred_sec = wallet.get_id_cred_sec(identity_index);
+    let id_cred_sec_hex = hex::encode(id_cred_sec);
+
+    let prf_key = wallet.get_prf_key(identity_index);
+    let prf_key_hex = hex::encode(prf_key);
+
+    let blinding_randomness = wallet.get_blinding_randomness(identity_index);
+    let blinding_randomness_hex = hex::encode(blinding_randomness);
+
+    let response = json!({
+        "idCredSec": id_cred_sec_hex,
+        "prfKey": prf_key_hex,
+        "blindingRandomness": blinding_randomness_hex
+    });
+    Ok(to_string(&response)?)
+}
+
+fn get_account_keys_and_randomness_aux(input: &str) -> anyhow::Result<String> {
+    let v: Value = from_str(&input)?;
+    let wallet = parse_wallet_input(&v)?;
+    let identity_index = try_get(&v, "identityIndex")?;
+    let account_credential_index = try_get(&v, "accountCredentialIndex")?;
+
+    let account_signing_key =
+        wallet.get_account_signing_key(identity_index, account_credential_index);
+    let account_signing_key_hex = hex::encode(account_signing_key);
+
+    let account_verify_key =
+        wallet.get_account_public_key(identity_index, account_credential_index);
+    let account_verify_key_hex = hex::encode(account_verify_key);
+
+    let mut attribute_commitment_randomness = HashMap::new();
+
+    for attribute_name in ATTRIBUTE_NAMES {
+        let attribute_tag = AttributeTag::from_str(attribute_name)?;
+        let commitment_randomness = wallet.get_attribute_commitment_randomness(
+            identity_index,
+            account_credential_index,
+            attribute_tag.0.into(),
+        );
+        let commitment_randomness_hex = hex::encode(commitment_randomness);
+        attribute_commitment_randomness.insert(attribute_tag.0, commitment_randomness_hex);
+    }
```

Comment:
I was debating the two options: return all the randomness, or let the caller send a list of attributes to get the randomness for. I went with the first because that was simpler on our end, but either way works for me. Let me know if you prefer them providing the list to get the attributes for.

In that regard, should I use the attribute_tag (i.e. the index 0, 1, 2, ...) or the text representation of the attributes?

Reply:
I think this option is fine. In any case it's us who'll have to handle it in a followup call to create_credential. This is fine.

I prefer string tags. Those are meant for JSON.